CMakeLists.txt

```diff
@@ -30,43 +31,53 @@ rosidl_generate_interfaces(${PROJECT_NAME}
   DEPENDENCIES sensor_msgs
 )
 
+rosidl_get_typesupport_target(cpp_typesupport_target "${PROJECT_NAME}" "rosidl_typesupport_cpp")
+
 # add the publisher example
 add_executable(my_publisher src/my_publisher.cpp)
-ament_target_dependencies(my_publisher
-  "cv_bridge"
-  "image_transport"
-  "OpenCV"
-  "rclcpp")
+target_link_libraries(my_publisher PRIVATE
+  cv_bridge::cv_bridge
+  image_transport::image_transport
+  opencv_imgcodecs
+  rclcpp::rclcpp
+)
 
 # add the subscriber example
 add_executable(my_subscriber src/my_subscriber.cpp)
-ament_target_dependencies(my_subscriber
-  "cv_bridge"
-  "image_transport"
-  "OpenCV"
-  "rclcpp")
+target_link_libraries(my_subscriber
+  cv_bridge::cv_bridge
+  image_transport::image_transport
+  opencv_highgui
+  rclcpp::rclcpp
+)
 
 # add the plugin example
 add_library(resized_plugins src/manifest.cpp src/resized_publisher.cpp src/resized_subscriber.cpp)
-ament_target_dependencies(resized_plugins
-  "image_transport"
-  "OpenCV")
+target_link_libraries(resized_plugins
+  "${cpp_typesupport_target}"
+  cv_bridge::cv_bridge
+  image_transport::image_transport
+  opencv_imgproc
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  "${sensor_msgs_TARGETS}"
+)
 
 # add the publisher from video example
 add_executable(publisher_from_video src/publisher_from_video.cpp)
-ament_target_dependencies(publisher_from_video
-  "cv_bridge"
-  "image_transport"
-  "OpenCV"
-  "rclcpp")
+target_link_libraries(publisher_from_video
+  cv_bridge::cv_bridge
+  image_transport::image_transport
+  opencv_highgui
+  opencv_videoio
+  rclcpp::rclcpp
+  "${sensor_msgs_TARGETS}"
+  "${std_msgs_TARGETS}"
+)
 
-# Install plugin descriptions
+# # Install plugin descriptions
```

Comment:
nitpick: this change looks like it could be reverted.

Reply:
Whoops, absolutely.  I had done that while debugging, I forgot to undo it.  Will fix.